dist/ExtUtils-ParseXS/lib/ExtUtils/ParseXS/Node.pm

```diff
@@ -1376,6 +1375,46 @@ sub as_boot_code {
     return [], \@lines;
 }
 
+# ======================================================================
+
+package ExtUtils::ParseXS::Node::TYPEMAP;
+
+# Process the lines associated with the TYPEMAP keyword
+#
+# fetch_para() will have already processed the <<EOF logic
+# and read all the lines up to, but not including, the EOF line.
+
+BEGIN { $build_subclass->(
+    'lines', # Array ref of all lines making up the TYPEMAP section
+)};
+
+
+# Feed all the lines to ExtUtils::Typemaps.
+
+sub parse {
+    my __PACKAGE__       $self = shift;
+    my ExtUtils::ParseXS $pxs  = shift;
+
+    $self->SUPER::parse($pxs); # set file/line_no
+
+    shift @{$pxs->{line}}; # skip the 'TYPEMAP:' line
+
+    # Suck in all remaining lines
+    $self->{lines} = [ @{$pxs->{line}} ];
+    @{$pxs->{line}} = ();
```

Comment:
Do we need all the copying here? ie.
```
$self->{lines} = $pxs->{line};
$pxs->{line} = [];
```
It's a bit icky reaching in the object... but we already do that by directly modifying `@{$pxs->{line}}`.

Mitigated by I expect most inline typemaps to be relatively small.